system/jlib/jsocket.cpp

```diff
@@ -4879,6 +4897,8 @@ class CSocketEpollThread: public CSocketBaseThread
 
     bool add(ISocket *sock,unsigned mode,ISocketSelectNotify *nfy)
     {
+        if (terminating)
```

Comment:
Does this need to be inside the critical block?

Reply:
a good idea, I have moved it